pandas/core/sparse/array.py

```diff
@@ -4,67 +4,82 @@
 from __future__ import division
 # pylint: disable=E1101,E1103,W0231
 
+import operator
+import numbers
 import numpy as np
 import warnings
 
 import pandas as pd
-from pandas.core.base import PandasObject, IndexOpsMixin
+import collections
+from pandas.core.base import PandasObject
 
 from pandas import compat
-from pandas.compat import range, PYPY
+from pandas.errors import PerformanceWarning
 from pandas.compat.numpy import function as nv
 
-from pandas.core.dtypes.generic import ABCSparseSeries
+from pandas.core.arrays.base import ExtensionArray, ExtensionOpsMixin
+import pandas.core.common as com
+from pandas.core.dtypes.generic import (
+    ABCSparseSeries, ABCSeries, ABCIndexClass
+)
 from pandas.core.dtypes.common import (
-    ensure_platform_int,
-    is_float, is_integer,
+    is_integer,
     is_object_dtype,
-    is_integer_dtype,
+    is_array_like,
+    pandas_dtype,
     is_bool_dtype,
     is_list_like,
     is_string_dtype,
     is_scalar, is_dtype_equal)
 from pandas.core.dtypes.cast import (
-    maybe_convert_platform, maybe_promote,
+    maybe_convert_platform,
     astype_nansafe, find_common_type, infer_dtype_from_scalar,
     construct_1d_arraylike_from_scalar)
 from pandas.core.dtypes.missing import isna, notna, na_value_for_dtype
+from pandas.core.missing import interpolate_2d
 
 import pandas._libs.sparse as splib
-import pandas._libs.lib as lib
-from pandas._libs.sparse import SparseIndex, BlockIndex, IntIndex
+from pandas._libs.sparse import BlockIndex, IntIndex
 from pandas._libs import index as libindex
 import pandas.core.algorithms as algos
-import pandas.core.ops as ops
 import pandas.io.formats.printing as printing
-from pandas.util._decorators import Appender
-from pandas.core.indexes.base import _index_shared_docs
+
+from .dtype import SparseDtype
```

Comment:
aren't we doing absolute imports?

Reply:
Probably (side note, I plan to move `sparse/array.py` to `array/sparse.py`, but that's a followup).